commit d86b99e60c187e00732af082b3f79b6d46bbf585
Author: Matthias Wolf <matthias.wolf@epfl.ch>
Date:   Mon Aug 9 14:36:58 2021 +0200

    dependencies: move lpips import into function
    
    This makes the lpips functionality somewhat optional by moving the
    import from the module scope into the function where it is used.  Should
    simplify installation.

diff --git a/README.md b/README.md
index e87f781..e7a29a0 100644
--- a/README.md
+++ b/README.md
@@ -80,7 +80,8 @@ All details related to installation and logic are described in the
 
 Some of the functionalities of `atlalign` depend on the [TensorFlow implementation
 of the Learned Perceptual Image Patch Similarity (LPIPS)](https://github.com/alexlee-gk/lpips-tensorflow). Unfortunately, the
-package is not available on PyPI and must be installed manually as follows.
+package is not available on PyPI and must be installed manually as follows
+for full functionality.
 ```shell script
 pip install git+http://github.com/alexlee-gk/lpips-tensorflow.git#egg=lpips_tf
 ```
diff --git a/atlalign/metrics.py b/atlalign/metrics.py
index 90563ac..4a0a6ed 100644
--- a/atlalign/metrics.py
+++ b/atlalign/metrics.py
@@ -38,18 +38,6 @@ from functools import wraps
 
 import ants
 
-try:
-    import lpips_tf
-except ModuleNotFoundError as err:
-    raise ModuleNotFoundError(
-        """
-        LPIPS-TensorFlow required but not found.
-
-        Please install it by running the following command:
-        $ pip install git+http://github.com/alexlee-gk/lpips-tensorflow.git#egg=lpips_tf
-        """
-    ) from err
-
 import numpy as np
 import pandas as pd
 import tensorflow as tf
@@ -996,6 +984,17 @@ def perceptual_loss_img(y_true, y_pred, model="net-lin", net="vgg"):
     We use the decorator just to make sure we do not run out of memory during a forward pass. Also,
     its fully convolutional but if the images are too small then might run into issues.
     """
+    try:
+        import lpips_tf
+    except ModuleNotFoundError as err:
+        raise ModuleNotFoundError(
+            """
+            LPIPS-TensorFlow required but not found.
+
+            Please install it by running the following command:
+            $ pip install git+http://github.com/alexlee-gk/lpips-tensorflow.git#egg=lpips_tf
+            """
+        ) from err
     # gray2rgb
     y_true = np.stack((y_true,) * 3, axis=-1)
     y_pred = np.stack((y_pred,) * 3, axis=-1)
